src/Agent.Listener/Configuration/AutoLogonConfigurationManager.cs

```diff
@@ -0,0 +1,258 @@
+#if OS_WINDOWS
+using Microsoft.VisualStudio.Services.Agent;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Security.Principal;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.Services.Agent.Listener.Configuration
+{
+    [ServiceLocator(Default = typeof(AutoLogonConfigurationManager))]
+    public interface IAutoLogonConfigurationManager : IAgentService
+    {
+        void Configure(CommandSettings command);
+        void Unconfigure();
+        bool RestartNeeded();
+    }
+
+    public class AutoLogonConfigurationManager : AgentService, IAutoLogonConfigurationManager
+    {
+        private ITerminal _terminal;
+        private INativeWindowsServiceHelper _windowsServiceHelper;
+        private IConfigurationStore _store;
+
+        public override void Initialize(IHostContext hostContext)
+        {
+            base.Initialize(hostContext);
+            _terminal = hostContext.GetService<ITerminal>();
+            _windowsServiceHelper = hostContext.GetService<INativeWindowsServiceHelper>();
+            _store = hostContext.GetService<IConfigurationStore>();
+        }
+
+        public void Configure(CommandSettings command)
+        {
+            if (!_windowsServiceHelper.IsRunningInElevatedMode())
+            {
+                Trace.Error("Needs Administrator privileges to configure agent with AutoLogon capability.");
+                Trace.Error("You will need to unconfigure the agent and then re-configure with Administrative rights");            
+                throw new SecurityException(StringUtil.Loc("NeedAdminForAutologonCapability"));
+            }
+
+            string domainName;
+            string userName;
+            string logonAccount;
+            string logonPassword;
+
+            while (true)
+            {   
+                logonAccount = command.GetWindowsLogonAccount(defaultValue: string.Empty, descriptionMsg: StringUtil.Loc("AutoLogonAccountNameDescription"));
```

Comment:
no it cant be.

---
In reply to: [118269221](https://github.com/Microsoft/vsts-agent/pull/985#discussion_r118269221) [](ancestors = 118269221)